### PR TITLE
Add --will-flag/destination/message/qos/retained options

### PR DESCRIPTION
**Pull request by zschwarz**
Don't know why, but log.severe("message") prints the message 2 times.

### CODE REVIEW

File: cli-paho-java/src/main/java/com/redhat/mqe/amc/Sender.java [hunk 1]
```diff
@@ -68,7 +68,24 @@ public void startClient() throws MqttException {
...
+                    log.severe("Will destination cannot be empty.");
+                    System.exit(0);
+                }
+
+                if (cliWillMessage.isEmpty()) {
+                    log.severe("Will message cannot be empty.");
+                    System.exit(0);
+                }
+
+                connectOptions.setWill(cliWillDestination, cliWillMessage.getBytes(), cliWillQos, cliWillRetained);
```
Comment (jdanekrh) [inline cli-paho-java/src/main/java/com/redhat/mqe/amc/Sender.java]: The will message relates to all three types of client (sender, receiver, (connector). Not only sender. Please add this to the receiver as well. Receivers can set last will message too.

File: cli-paho-java/src/main/java/com/redhat/mqe/amc/Client.java [hunk 2]
```diff
@@ -99,6 +109,16 @@ OptionParser populateOptionParser(OptionParser parser) {
         logMsgs = parser.accepts("log-msgs", "print messages").withRequiredArg()
             .ofType(String.class).defaultsTo("none");
 
+        willFlag = parser.accepts("will-flag", "will flag (0,1) ").withRequiredArg().ofType(Integer.class).defaultsTo(0);
```
Comment (jdanekrh) [inline cli-paho-java/src/main/java/com/redhat/mqe/amc/Client.java]: why not boolean type?

File: cli-paho-java/src/main/java/com/redhat/mqe/amc/Client.java [hunk 3]
```diff
@@ -99,6 +109,16 @@ OptionParser populateOptionParser(OptionParser parser) {
...
+
+        willMessage = parser.accepts("will-message", "will message").withRequiredArg().ofType(String.class).defaultsTo("");
```
Comment (jdanekrh) [inline cli-paho-java/src/main/java/com/redhat/mqe/amc/Client.java]: will message _body_

File: cli-paho-java/src/main/java/com/redhat/mqe/amc/Client.java [hunk 4]
```diff
@@ -99,6 +109,16 @@ OptionParser populateOptionParser(OptionParser parser) {
...
+
+        willQos = parser.accepts("will-qos", "will QoS (0,1,2) ").withRequiredArg().ofType(Integer.class).defaultsTo(0);
+
+        willRetained = parser.accepts("will-retained", "is will retained").withRequiredArg().ofType(Boolean.class).defaultsTo(false);
+
+        willDestination = parser.accepts("will-destination", "will topic name").withRequiredArg().ofType(String.class).defaultsTo("");
```
Comment (jdanekrh) [inline cli-paho-java/src/main/java/com/redhat/mqe/amc/Client.java]: following the naming of other options, why not --conn-will-destination, etc. The will message is related to connection, IMO.

File: cli-paho-java/src/main/java/com/redhat/mqe/amc/Client.java (hunk 2 above)
Comment (rvais) [reply cli-paho-java/src/main/java/com/redhat/mqe/amc/Client.java]: Wanted to ask the same thing

**Pull request merged**